android/cli/lib/process-drawables-task.js

```diff
@@ -0,0 +1,74 @@
+const path = require('path');
+const CopyResourcesTask = require('../../../cli/lib/tasks/copy-resources-task');
+
+const appc = require('node-appc');
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+
+const drawableDpiRegExp = /^(high|medium|low)$/;
+
+/**
+ * Task that copies Android drawables into the app.
+ */
+class ProcessDrawablesTask extends CopyResourcesTask {
+
+	/**
+	 * Constructs a new processing task.
+	 *
+	 * @param {Object} options Configuration object for this task.
+	 * @param {String} [options.name='process-drawables'] Name for the task.
+	 * @param {String} options.incrementalDirectory Path to a folder where incremental task data will be stored.
+	 * @param {Map<string,FileInfo>} options.files Map of input files to file info
+	 * @param {Object} [options.logger] The logger instance used by this task.
+	 * @param {Object} options.builder Builder instance.
+	 */
+	constructor(options) {
+		options.name = options.name || 'process-drawables';
+		const builder = options.builder;
+		const logger = builder.logger;
+		const appMainResDir = builder.buildAppMainResDir;
+
+		// We have to modify the destination paths for each file. Is there a place we can move this to other than constructor?
+		// Because we want the expected final filename for whenever the incremental task stuff needs it
+		options.files.forEach((info, relPath) => {
+			const parts = relPath.split(path.sep);
+			const origFoldername = parts[1];
+			const foldername = drawableDpiRegExp.test(origFoldername) ? 'drawable-' + origFoldername[0] + 'dpi' : 'drawable-' + origFoldername.substring(4);
+
+			const destFilename = `${info.name}.${info.ext}`;
+
+			// We have a drawable image file. (Rename it if it contains invalid characters.)
+			const warningMessages = [];
+			if (parts.length > 3) {
+				warningMessages.push(__('- Files cannot be put into subdirectories.'));
+			}
+			// info.name may have .9 suffix, if so, we do not want to convert that .9 to _9
+			let base = info.name;
+			let suffix = '';
+			if (info.name.endsWith('.9')) {
+				base = info.name.substring(0, info.name.length - 2);
+				suffix = '.9';
+			}
```

Comment:
We have a major regression here.

We're not correctly renaming "res" files put under "subdirectories" like the below.
`./Resources/android/images/res-mdpi/logos/app.png`

The above file should be renamed to the below. Effectively, path separators should turn into underscores.
`logos_app.png`

But currently the code is stripping away the subdirectory name like the below. This prevents the Titanium app from finding the image. _(Note that many noteworthy Titanium apps put res files under subdirectories, even though we log warnings about it.)_
`app.png`

We should probably do something like the below.
```javascript
let base = info.name;
let suffix = '';
if (info.name.endsWith('.9')) {
	base = info.name.substring(0, info.name.length - 2);
	suffix = '.9';
}
if (parts.length > 3) {
	warningMessages.push(__('- Files cannot be put into subdirectories.'));
	base = parts.slice(3, parts.length - 1).join('_') + '_' + base;
}
let destFilteredFilename = `${base.toLowerCase().replace(/[^a-z0-9_]/g, '_')}${suffix}.${info.ext}`;
```